homeassistant/components/home_connect/__init__.py

```diff
@@ -39,9 +65,55 @@
     extra=vol.ALLOW_EXTRA,
 )
 
+SERVICE_SETTING_SCHEMA = vol.Schema(
+    {
+        vol.Required(ATTR_DEVICE_ID): str,
+        vol.Required(ATTR_KEY): str,
+        vol.Required(ATTR_VALUE): vol.Any(str, int, bool),
+    }
+)
+
+SERVICE_OPTION_SCHEMA = vol.Schema(
+    {
+        vol.Required(ATTR_DEVICE_ID): str,
+        vol.Required(ATTR_KEY): str,
+        vol.Required(ATTR_VALUE): vol.Any(str, int, bool),
+        vol.Optional(ATTR_UNIT): str,
+    }
+)
+
+SERVICE_PROGRAM_SCHEMA = vol.Any(
+    {
+        vol.Required(ATTR_DEVICE_ID): str,
+        vol.Required(ATTR_PROGRAM): str,
+        vol.Required(ATTR_KEY): str,
+        vol.Required(ATTR_VALUE): vol.Any(int, str),
+        vol.Optional(ATTR_UNIT): str,
+    },
+    {
+        vol.Required(ATTR_DEVICE_ID): str,
+        vol.Required(ATTR_PROGRAM): str,
+    },
+)
+
+SERVICE_COMMAND_SCHEMA = vol.Schema({vol.Required(ATTR_DEVICE_ID): str})
+
 PLATFORMS = [Platform.BINARY_SENSOR, Platform.LIGHT, Platform.SENSOR, Platform.SWITCH]
 
 
+def _get_appliance_by_device_id(
+    hass: HomeAssistant, device_id: str
+) -> api.HomeConnectDevice | None:
+    """Return a Home Connect appliance instance given an device_id."""
+    for hc_api in hass.data[DOMAIN].values():
+        for dev_dict in hc_api.devices:
+            device = dev_dict[CONF_DEVICE]
+            if device.device_id == device_id:
+                return device.appliance
+    _LOGGER.error("Appliance for device id %s not found", device_id)
```

Comment:
We prefer raising either ValueError on bad user input or HomeAssistantError if the api call fails in service handlers.